### PR TITLE
add copySortSource feature

### PR DESCRIPTION
**Pull request by beatfreaker**
PR for [#74](https://github.com/bevacqua/dragula/issues/74)

This features allow to move the copyable div in same container to reorder its position if copySortSource option is set to true.

### CODE REVIEW

**Comment by beatfreaker** [comment]
@bevacqua Please review this.

File: dragula.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ function dragula (initialContainers, options) {
   var _renderTimer; // timer for setTimeout renderMirrorImage
   var _lastDropTarget = null; // last container item was over
   var _grabbed; // holds mousedown context until first mousemove
+  var _copySortSource; // if true then allow ordering the copyable div in same container
```
Comment (bevacqua) [inline dragula.js]: Just use `o.copySortSource` directly

File: dragula.js [hunk 2]
```diff
@@ -49,6 +51,10 @@ function dragula (initialContainers, options) {
   if (o.removeOnSpill === true) {
     drake.on('over', spillOver).on('out', spillOut);
   }
+  
+  if (o.copySortSource === true) {
+    _copySortSource = true;
```
Comment (bevacqua) [inline dragula.js]: Unnecessary if used directly

File: dragula.js [hunk 3]
```diff
@@ -218,6 +224,10 @@ function dragula (initialContainers, options) {
   }
 
   function drop (item, target) {
+    var parent = item.parentElement;
+	if (_copy && _copySortSource && target === _source) {
```
Comment (bevacqua) [inline dragula.js]: Please use two spaces for indentation

**Comment by bevacqua** [comment]
Great work. Thanks! Please fix the issues I commented and squash your commits and I'll merge. Thanks

**Comment by beatfreaker** [comment]
@bevacqua did the changes please check and merge it or if required any other changes then let me know.

**Comment by beatfreaker** [comment]
@bevacqua is this good to merge??

**Comment by bevacqua** [comment]
> This branch has conflicts that must be resolved

**Comment by beatfreaker** [comment]
@bevacqua check now

**Comment by bevacqua** [comment]
Please rebase from `master`

File: dist/dragula.js [hunk 4]
```diff
@@ -372,7 +380,8 @@ function dragula (initialContainers, options) {
       _lastDropTarget = dropTarget;
       over();
     }
-    if (dropTarget === _source && _copy) {
+	console.log(!_copySortSource);
```
Comment (nescalante) [inline dist/dragula.js]: missing `console.log` here

File: dist/dragula.js [hunk 5]
```diff
@@ -255,6 +259,10 @@ function dragula (initialContainers, options) {
   }
 
   function drop (item, target) {
+    var parent = item.parentElement;
+	if (_copy && _copySortSource && target === _source) {
```
Comment (nescalante) [inline dist/dragula.js]: missed identation for this `if` statement

File: dist/dragula.js [hunk 6]
```diff
@@ -86,6 +87,10 @@ function dragula (initialContainers, options) {
   if (o.removeOnSpill === true) {
     drake.on('over', spillOver).on('out', spillOut);
   }
+  
+  if(o.copySortSource === true) {
```
Comment (nescalante) [inline dist/dragula.js]: we can change this to just `_copySortSource = o.copySortSource` (and remove the `_copySortSource` assignment from line 76)

**Comment by bevacqua** [comment]
Oh and also please reset changes to `dist`. Those shouldn't be part of the PR. See [contributing guidelines](https://github.com/bevacqua/dragula/blob/master/contributing.markdown#build)

**Comment by beatfreaker** [comment]
Closng this PR .... refer this one [#203](https://github.com/bevacqua/dragula/pull/203)

**Pull request closed without merging**